app/models/training_period.rb

```diff
@@ -45,6 +45,10 @@ class TrainingPeriod < ApplicationRecord
   scope :for_ect, ->(ect_at_school_period_id) { where(ect_at_school_period_id:) }
   scope :for_mentor, ->(mentor_at_school_period_id) { where(mentor_at_school_period_id:) }
   scope :for_school_partnership, ->(school_partnership_id) { where(school_partnership_id:) }
+  scope :at_school, ->(school_id) {
+    left_joins(:ect_at_school_period, :mentor_at_school_period)
+      .where('ect_at_school_periods.school_id = :id OR mentor_at_school_periods.school_id = :id', id: school_id)
+  }
```

Comment:
I think this can be done without resorting to SQL strings.

There's a scope `MentorAtSchoolPeriod.for_school` that could(should) have a corresponding scope on `ECTAtSchoolPeriod`.

With those in place, this query could be:

```ruby
Teacher.left_outer_joins(:ect_at_school_periods, :mentor_at_school_periods)
        .merge(
          MentorAtSchoolPeriod.for_school(school).or(ECTAtSchoolPeriod.for_school(school))
        )

```